### ISSUE TITLE
Error documentation for dynamic field mapping with JSON string type

### ISSUE THREAD

**Issue by RickyLau** [elastic/elasticsearch#66910]
**Elasticsearch version** (`bin/elasticsearch --version`):7.10.1

**Plugins installed**: []

**JVM version** (`java -version`):  
``` 
  openjdk version "15.0.1" 2020-10-20
  OpenJDK Runtime Environment AdoptOpenJDK (build 15.0.1+9)
  OpenJDK 64-Bit Server VM AdoptOpenJDK (build 15.0.1+9, mixed mode, sharing)
```

**OS version** (`uname -a` if on a Unix-like system):
```
  Linux es-21 3.10.0-957.el7.x86_64 #1 SMP Thu Nov 8 23:39:32 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```
**Description of the problem including expected versus actual behavior**:
This is a documentation error from Mapping-> Dynamic Mapping-> dynamic-field-mapping. If the number detection function is enabled, when the floating-point value in the JSON is passed, the field mapping should map this type to Elasticsearch double value, but in fact, it is a float mapping. 
In this case, I think the document is out of date.
**Steps to reproduce**:

 1. Create indices & enable number detection
 ```
  PUT test_numberic_detection
  {
    "mappings": {
      "numeric_detection": true
    }
  }
 ```
 2. Insert a document with floating-point value.
 ```
  POST test_numberic_detection/_doc
  {
    "price":"2.12"
  }
 ```
 3. Check dynamic filed mapping result
 ```
 GET test_numberic_detection/_mapping?pretty
 ```
4. The result is as follows
 ```
  {
    "test_numberic_detection" : {
      "mappings" : {
        "numeric_detection" : true,
        "properties" : {
          "price" : {
            "type" : "float"
          }
        }
      }
    }
  }
 ```

**Comment by jtibshirani** [comment]
The current docs seem correct to me -- in dynamic mappings, JSON floating point values are mapped to `float`. The `numeric_detection` flag applies to JSON strings, allowing them to be mapped as numeric types if the values can be parsed. Is there a way in which you think these docs could be clarified?

**Comment by RickyLau** [comment]
> The current docs seem correct to me -- in dynamic mappings, JSON floating point values are mapped to `float`. The `numeric_detection` helps handle JSON strings, allowing them to be mapped as numeric types if the values can be parsed. Is there a way in which you think these docs could be clarified?

The current or master document maps the JSON double type to the elasticsearch double type, but actually maps the JSON double type to the elasticsearch float type.

**Comment by jtibshirani** [comment]
Thanks, I see the problem now. I agree this table should be fixed:

![Screen Shot 2021-01-04 at 6 06 09 PM](https://user-images.githubusercontent.com/7461306/103598477-8c9ccc80-4eb7-11eb-8561-c03104160608.png)

**Comment by RickyLau** [comment]
@jtibshirani  Thank you for your help.

**Comment by jrodewig** [comment]
This has been closed with #66530. If I've missed something, please feel free to re-open. Thanks for raising this @RickyLau.
